src/snap/install.ts

```diff
@@ -0,0 +1,88 @@
+import { MetaMaskInpageProvider } from "@metamask/providers";
+import { Page } from "puppeteer";
+import {
+  clickOnButton,
+  clickOnElement,
+  clickOnLogo,
+  openProfileDropdown,
+} from "../helpers";
+import { flaskOnly } from "./utils";
+
+declare let window: { ethereum: MetaMaskInpageProvider };
+
+export type InstallStep = (page: Page) => Promise<void>;
+
+export async function installSnap(
+  page: Page,
+  snapId: string,
+  opts: {
+    hasPermissions: boolean;
+    hasKeyPermissions: boolean;
+    customSteps?: InstallStep[];
+    version?: string;
+  }
+): Promise<void> {
+  flaskOnly(page);
+  //need to open page to access window.ethereum
+  const installPage = await page.browser().newPage();
+  await installPage.goto("https://google.com");
+  await installPage.evaluate(
+    (opts: { snapId: string; version?: string }) => {
+      void window.ethereum.request({
```

Comment:
I wonder if there is a way to get the result of this method as then we wouldn't need to check if the snap is installed. Currently, if you await this, you cannot click the confirmation prompts

I need to try using https://www.npmjs.com/package/p-defer to evaluate and resolve with the result.

Reply:
You can return inside evaluate function, as result of it https://pptr.dev/api/puppeteer.page.evaluate/#example-1